src/gluonts/mx/trainer/callback.py

```diff
@@ -355,25 +362,96 @@ def __init__(self, time_limit: Optional[int] = None) -> None:
             time in seconds, after which training ends
         """
         self.time_limit = time_limit
-        self.start_time = -1.0
+        self.include_validation_in_time_limit = (
+            include_validation_in_time_limit
+        )
+        self.stop_during_epoch = stop_during_epoch
+        self.time_spent = 0.0
+        self.checkpoint = -1.0
 
     def on_train_start(self, max_epochs: int) -> None:
-        self.start_time = time.time()
+        self.checkpoint = time.time()
+
+    def on_train_batch_end(self, training_network: nn.HybridBlock) -> bool:
+        print(
+            "on_train_batch_end", self.time_spent
+        )  # for debugging purpose, will be deleted before merging
+        self.time_spent += time.time() - self.checkpoint
+        self.checkpoint = time.time()
+        if self.stop_during_epoch:
+            if self.time_spent > self.time_limit:
```

Comment:
Log for consistency?

Reply:
I think we don't need log here, otherwise, the log will be printed twice(one after the batch, and one after the epoch). Because regardless of whether we are to stop after one batch, time limit will always be check after one epoch.